pkg/log/context.go

```diff
@@ -36,3 +58,40 @@ func goid() string {
 	idField := strings.Fields(strings.TrimPrefix(string(buf[:n]), "goroutine "))[0]
 	return idField
 }
+
+func paramKey(name string) string {
+	return PARAM + name
+}
+
+func GetParamFromContext(ctx context.Context, key string) (string, string) {
+	val := ctx.Value(contextKey(paramKey(key)))
+	if val == nil {
+		return key, ""
+	} else {
+		return key, val.(string)
+	}
+}
+
+//openapi middleware handler that takes URL params from the path and put them
+//on the context for further use (mainly by logs). parameters are prefixed to avoid
+//conflicts (for example: cluster_id --> PARAM$cluster_id)
+func ContextHandler() func(http.Handler) http.Handler {
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//extract the Route from the request
+			mr := middleware.MatchedRouteFrom(r)
+			//set the parameters on the request's context
+			ctx := r.Context()
```

Comment:
This should be done only if mr is not nil

Reply:
DONE